pkg/scheduler/scheduler.go

```diff
@@ -402,3 +399,33 @@ func (depGraph DependencyGraph) GetStatus() (interfaces.DeploymentStatus, interf
 	}
 	return status, deploymentReport
 }
+
+func runConcurrently(funcs []func() bool, concurrency int) bool {
```

Comment:
My proposal for this function:
```
func runConcurrently(funcs []func() bool, concurrency int) bool {
	sem := make(chan bool, concurrency)
	defer close(sem)

	result := true

	for _, f := range funcs {
		sem <- true
		go func(foo func() bool) {
			defer func() { <-sem }()
			if !foo() {
				result = false
			}
		}(f)
	}

	for i := 0; i < cap(sem); i++ {
		sem <- true
	}
	return result
}
```

Check out how it works here: https://play.golang.org/p/yVQ7lqXT7W